config/settings/base.py

```diff
@@ -201,6 +201,19 @@
 # https://timonweb.com/django/overriding-field-widgets-in-django-doesnt-work-template-not-found-the-solution/
 FORM_RENDERER = "django.forms.renderers.TemplatesSetting"
 
+try:
+    # This module is only available when running under uWSGI
+    # https://uwsgi-docs.readthedocs.io/en/latest/PythonModule.html
+    import uwsgi  # noqa: F401
+except ImportError:
+    uwsgi_db_statement_timeout = None
+    uwsgi_db_lock_timeout = None
+else:
+    uwsgi_db_statement_timeout = os.environ.get("UWSGI_DB_STATEMENT_TIMEOUT", 10_000)
+    uwsgi_db_lock_timeout = os.environ.get("UWSGI_DB_LOCK_TIMEOUT", 5_000)
+db_statement_timeout = int(uwsgi_db_statement_timeout or os.getenv("DB_STATEMENT_TIMEOUT", 0))
+db_lock_timeout = int(uwsgi_db_lock_timeout or os.getenv("DB_LOCK_TIMEOUT", 0))
```

Comment:
Je simplifierais la logique :
```suggestion
except ImportError:
    db_statement_timeout = int(uwsgi_db_statement_timeout or os.getenv("DB_STATEMENT_TIMEOUT", 0))
    db_lock_timeout = int(uwsgi_db_lock_timeout or os.getenv("DB_LOCK_TIMEOUT", 0))
else:
    db_statement_timeout = os.environ.get("UWSGI_DB_STATEMENT_TIMEOUT", 10_000)
    db_lock_timeout = os.environ.get("UWSGI_DB_LOCK_TIMEOUT", 5_000)
```

Reply:
En voyant les deux, je préfère la solution 1., pour permettre de désactiver le statement timeout pour uWSGI.
Autrement, faudrait définir `UWSGI_DB_STATEMENT_TIMEOUT=0` **et** `DB_STATEMENT_TIMEOUT=0`.
Avoir deux settings séparé rend la configuration plus claire.

Reply:
Dans la solution 2 tu n'as que `UWSGI_DB_STATEMENT_TIMEOUT=0` à définir pour la désactivation vu que ça teste `is not None` et pas si c'est _falsy_.

J'ai proposé 1 car c'était la logique que j'avais comprise et donc je me suis dit qu'on voulais 2 manettes complètement distincte et pas une manette spécialisation de la précédente (la métaphore claqué au sol :grin:), qui pour le coup est ce qui me semble plus naturelle et logique d'un point de vue configuration et administration.
Bon après je dit ça mais à la fin les deux seront configurés dans le même fichier car je doute qu'on veuille avoir la même valeur et donc ne garder que `DB_STATEMENT_TIMEOUT`,  au final le fonctionnement sera celui de la solution 1...

Et je renommerais bien `UWSGI_DB_STATEMENT_TIMEOUT` en `WWW_DB_STATEMENT_TIMEOUT`, comme ça l'intuition est claire et agnostique ;).